### PR TITLE
Delete features which are easily removed, in libsyntax

### PR DESCRIPTION
**Pull request by strega-nil**

### CODE REVIEW

**Comment by aidanhs** [comment]
Hi @ubsan, looks like this has failed travis because the Cargo.lock needs updating as well?

**Comment by ubsan** [comment]
@aidanhs ah, dang

**Comment by ubsan** [comment]
Hopefully that fixes it.

File: src/libsyntax/parse/mod.rs [hunk 1]
```diff
@@ -261,10 +261,14 @@ pub fn char_lit(lit: &str) -> (char, isize) {
     }
 }
 
+pub fn escape_default(s: &str) -> String {
+  s.chars().map(char::escape_default).flat_map(|x| x).collect()
```
Comment (emilio) [inline src/libsyntax/parse/mod.rs]: nit: indentation is weird here.

Comment (ubsan) [inline src/libsyntax/parse/mod.rs]: oh, right, I use a different style from rustc.

**Comment by kennytm** [comment]
Could some features from `libsyntax_pos` (a dependency of `libsyntax`) be removed as well?

```sh
/rust/src/libsyntax_pos$ git grep '\[feature'
lib.rs:26:#![feature(const_fn)]
lib.rs:27:#![feature(custom_attribute)]
lib.rs:28:#![feature(optin_builtin_traits)]
lib.rs:30:#![feature(rustc_private)]
lib.rs:31:#![feature(staged_api)]
lib.rs:32:#![feature(specialization)]
```

**Comment by ubsan** [comment]
@kennytm it's a lot harder.

**Comment by kennytm** [comment]
@ubsan I see, thanks. I thought they are pulled in also due to bitflags.

**Comment by Mark-Simulacrum** [comment]
I think this is done, though commits could be squashed.

**Comment by ubsan** [comment]
@Mark-Simulacrum done

**Pull request merged**